### PR TITLE
boards/nucleo-f429zi: Provide 2nd timer

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

- Add a common timer config with two (instead of only one) timer using TIM2 + TIM5
    - Mostly copy-pasting the cfg_timer_tim2.h and cfg_timer_tim5.h together
- Make use of that for the `nucleo-f429zi`

### Testing procedure

E.g. `tests/periph_timer`, but also grepping for TIM2 and TIM5 in `boards/nucleo-f429zi/include/periph_conf.h` to detect any conflict e.g. between PWM and timer config.

### Issues/PRs references

None

### CODE REVIEW

**Comment by gschorcht** [comment]
Would this change also be valid for `stm32f429i-disc1`?

File: boards/common/stm32/include/cfg_timer_tim2_and_tim5.h [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        .rcc_mask = RCC_MC_APB1ENSETR_TIM2EN,
+#else
+        .rcc_mask = RCC_APB1ENR_TIM2EN,
+#endif
+        .bus      = APB1,
+        .irqn     = TIM2_IRQn
+    },
+    {
+        .dev      = TIM5,
+        .max      = 0xffffffff,
```
Comment (kaspar030) [inline boards/common/stm32/include/cfg_timer_tim2_and_tim5.h:54]: Is there any reason for the sorting?
If TIM5 is always available and always 32bit, it should be the first in the array, to trickle down as default for ztimer_used.

Comment (maribu) [reply boards/common/stm32/include/cfg_timer_tim2_and_tim5.h:54]: At least on the L4 it indeed seems to be available and 32 bit. I will swap the two.

**Comment by maribu** [comment]
> Would this change also be valid for `stm32f429i-disc1`?

Likely, but I don't have the hardware to test. I intentionally split out the timer config as separate header file so that using it for other boards is a one-line-diff PR.

File: boards/common/stm32/include/cfg_timer_tim2_and_tim5.h (hunk 1 above)
Comment (maribu) [reply boards/common/stm32/include/cfg_timer_tim2_and_tim5.h]: https://github.com/RIOT-OS/RIOT/blob/1961274705d75aee1c7482c19493b053740de150/boards/nucleo-l152re/doc.txt#L29

Yep, at least on STM32L1 the sorting helps ztimer

**Comment by maribu** [comment]
Maybe I should also call the file tim5_and_tim2 then with the new sorting

**Comment by maribu** [comment]
CI is hopefully happy now :)

**Comment by gschorcht** [comment]
> > Would this change also be valid for `stm32f429i-disc1`?
> 
> Likely, but I don't have the hardware to test. I intentionally split out the timer config as separate header file so that using it for other boards is a one-line-diff PR.

I could test it, I have the hardware. But since `stm32f429i-disc` doesn't provide a PWM config, we don't have to expect conflicts.

**Comment by maribu** [comment]
All seems to be green :)

**Review by benpicco — APPROVED** [review]
bors merge

**Comment by maribu** [comment]
Thx :)

**Pull request merged**